lib/sqlalchemy/sql/schema.py

```diff
@@ -5664,3 +5707,239 @@ def _copy(self, **kw: Any) -> Identity:
         )
 
         return self._schema_item_copy(i)
+
+
+class Period(SchemaItem):
+    """Defines a period construction, i.e. ``PERIOD FOR period_name
+    (start_col, end_col)`` syntax.
+
+    .. versionadded:: TODO
+    """
+
+    __visit_name__ = "period"
+
+    table: Table
+    start: Column
+    end: Column
+    system: bool
+
+    def __init__(
+        self,
+        name: str,
+        start: _DDLColumnArgument,
+        end: _DDLColumnArgument,
+        system: Optional[bool] = False,
+        primary_key: Optional[bool] = False,
+    ) -> None:
+        """Construct a ``PERIOD FOR`` DDL construct on two columns.
+
+        :param name:
+            The name to give the period.
+
+        :param start:
+            A target column that defines the period's start. A
+            :class:`_schema.Column` object or a column name as a string.
+
+        :param end:
+            A target column that defines the period's start. A
+            :class:`_schema.Column` object or a column name as a string.
+
+        :param system:
+            Indicates that the ``Period`` is implicit, i.e. should not be
+            generated in DDL.
+
```

Comment:
**Federico Caselli** (CaselIT) wrote:

this seems more a comment, but it's fine here too

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3803